### ISSUE TITLE
Package with the same name as source package is not imported

### ISSUE THREAD

**Issue by noodlensk** [matryer/moq#61]
Source:
```GO
package mypackage

import anotherpkg "github.com/noodlensk/moq/issues/repo1/mypackage"

//go:generate moq -out client_moq.go . Client
type Client interface {
	GetSomeThing() anotherpkg.MyStruct
}
```
Result mock:
```GO
package mypackage

import (
	"sync"
)
...
type ClientMock struct {
	// GetSomeThingFunc mocks the GetSomeThing method.
	GetSomeThingFunc func() MyStruct
...
}
```
Expected mock:
```GO
package mypackage

import (
	"sync"
        anotherpkg "github.com/noodlensk/moq/issues/repo1/mypackage"
)
...
type ClientMock struct {
	// GetSomeThingFunc mocks the GetSomeThing method.
	GetSomeThingFunc func() anotherpkg.MyStruct
...
}
```

I've checked *moq* source code, I guess it should be fixed by:
- Support packages aliases in import section
- Compare packages paths instead of packages names

**Comment by pdrum** [comment]
I just faced the same issue

**Comment by matryer** [comment]
OK I'll see if I can figure out what's going on.

**Comment by matryer** [comment]
@jasonhancock @pdrum @noodlensk Can you please update and check this again? #62 may have fixed it.

**Comment by pdrum** [comment]
I checked it and it seems to have been fixed.

### Input 
```go
package sample

import "sandbox/sample/sample"

//go:generate moq . Interface
type Interface interface {
	DoThing(number sample.Number)
}
```
 and 
```go
package sample

type Number int

```

### Output
```go
// Code generated by moq; DO NOT EDIT.
// github.com/matryer/moq

package sample

import (
	"sandbox/sample/sample"
	"sync"
)

var (
	lockInterfaceMockDoThing sync.RWMutex
)

// InterfaceMock is a mock implementation of Interface.
//
//     func TestSomethingThatUsesInterface(t *testing.T) {
//
//         // make and configure a mocked Interface
//         mockedInterface := &InterfaceMock{
//             DoThingFunc: func(number sample.Number)  {
// 	               panic("TODO: mock out the DoThing method")
//             },
//         }
//
//         // TODO: use mockedInterface in code that requires Interface
//         //       and then make assertions.
//
//     }
type InterfaceMock struct {
	// DoThingFunc mocks the DoThing method.
	DoThingFunc func(number sample.Number)

	// calls tracks calls to the methods.
	calls struct {
		// DoThing holds details about calls to the DoThing method.
		DoThing []struct {
			// Number is the number argument value.
			Number sample.Number
		}
	}
}

// DoThing calls DoThingFunc.
func (mock *InterfaceMock) DoThing(number sample.Number) {
	if mock.DoThingFunc == nil {
		panic("InterfaceMock.DoThingFunc: method is nil but Interface.DoThing was just called")
	}
	callInfo := struct {
		Number sample.Number
	}{
		Number: number,
	}
	lockInterfaceMockDoThing.Lock()
	mock.calls.DoThing = append(mock.calls.DoThing, callInfo)
	lockInterfaceMockDoThing.Unlock()
	mock.DoThingFunc(number)
}

// DoThingCalls gets all the calls that were made to DoThing.
// Check the length with:
//     len(mockedInterface.DoThingCalls())
func (mock *InterfaceMock) DoThingCalls() []struct {
	Number sample.Number
} {
	var calls []struct {
		Number sample.Number
	}
	lockInterfaceMockDoThing.RLock()
	calls = mock.calls.DoThing
	lockInterfaceMockDoThing.RUnlock()
	return calls
}

```

**Comment by jasonhancock** [comment]
Looks good on my end as well

**Comment by sudo-suhas** [comment]
Closing as this specific issue was solved in #62.
